### PR TITLE
Make the width of every slice dynamic according to the slice value

### PR DESCRIPTION
**Pull request by sarykemmoun**
Hello,
Your library is really useful.
I added a type of pie that every slice has its own width according to its value
It depends on a new prop - calls 'dinamicWidth' (default false)
Attached a sample of this type. 

`<PieChart
                style={{
                    flex: 1,
                    width: pieChartSize
                }}
                innerRadius={'40%'}
                outerRadius={'80%'}
                dinamicWidth
                padAngle={0}
                data={pieData}
            />`

screenshot from 2018-02-27 15-56-37

### CODE REVIEW

**Comment by JesperLekland** [comment]
Hi @sarykemmoun and thank you for your contribution. I really like your suggestion 🙂 I would however like to have something a bit more dynamic. There will be plenty of use cases where users will want to size the slices themselves (e.g when somebody clicks a slice). If you can think of a good way of supporting this that would be great!

File: src/pie-chart.js [hunk 1]
```diff
@@ -35,20 +35,29 @@ class PieChart extends PureComponent {
             return defaultVal
         }
     }
+    _getDinamicOuterRadiuses(index, length, max) {
```
Comment (JesperLekland) [inline src/pie-chart.js]: spelling -> _getDynamicRadius

File: src/pie-chart.js [hunk 2]
```diff
@@ -35,20 +35,29 @@ class PieChart extends PureComponent {
...
+        let { innerRadius, outerRadius } = this.props
+        if (typeof innerRadius === 'string') innerRadius = (innerRadius.split('%')[0] / 100) * max
```
Comment (JesperLekland) [inline src/pie-chart.js]: don't do one line `if`. Will update lint file to reflect this preference

File: src/pie-chart.js [hunk 3]
```diff
@@ -75,23 +84,31 @@ class PieChart extends PureComponent {
             console.warn('innerRadius is equal to or greater than outerRadius')
         }
 
-        const arc = shape.arc()
+        let arc = shape.arc()
```
Comment (JesperLekland) [inline src/pie-chart.js]: try to avoid `let`, will update lint with this as well

File: src/pie-chart.js [hunk 4]
```diff
@@ -75,23 +84,31 @@ class PieChart extends PureComponent {
...
+                .innerRadius(_labelRadius)
+                .padAngle(padAngle) :
+            arc
 
         const pieSlices = shape.pie()
             .value(d => d.value)
             .sort(sort)
             (data)
 
+        const getDinamicAcr = (index, length) => {
```
Comment (JesperLekland) [inline src/pie-chart.js]: spelling

File: src/pie-chart.js [hunk 5]
```diff
@@ -35,20 +35,29 @@ class PieChart extends PureComponent {
...
+            innerRadius,
+            outerRadius,
+            labelRadius,
+            padAngle,
+            animate,
+            animationDuration,
+            style,
+            renderDecorator,
+            sort,
+            dinamicWidth,
```
Comment (JesperLekland) [inline src/pie-chart.js]: spelling, `dynamicWidth`

**Comment by sarykemmoun** [comment]
Hello @JesperLekland. First, thank you for the response.
For making the feature more dynamic we should consider adding a prop of 'renderSlice' that it'll be an option prop of the data array.
 In that way someone that would like, for example, to make the pressed slice bigger, he'll change this prop on 'onPress', so the code of path component would be 
(<Path
         key={key}
          fill={color}
          **d={renderSlice||arc(slice)}**
          animate={animate}
          animationDuration={animationDuration}
          onPress={onPress}
     />)
But I'm afraid that it becomes too dynamic since the user would have to understand well and use 
the libraries react-native-svg and d3-shape.
What do you think about it?

**Comment by JesperLekland** [comment]
Hi again,

I've implemented a more generic solution in v4.0.0. It doesn't automatically size the slices but allows the user to do it manually, hopefully it will solve your problem. Check out the second example [here](https://github.com/JesperLekland/react-native-svg-charts-examples#piechart)

I'll go ahead and close this PR

**Pull request closed without merging**